pymeasure/instruments/rohdeschwarz/sfm.py

```diff
@@ -0,0 +1,1501 @@
+        values=[32E6, 46E6],
+        )
+
+    vision_average = Instrument.control(
+        "SOUR:TEL:MOD:VIS:AVER:STAT?",
+        "SOUR:TEL:MOD:VIS:AVER:STAT %s",
+        """ A bool property that controls the average mode for the vision system
+
+        ======  =======
+        Value   Meaning
+        ======  =======
+        False   Average function disabled
+        True    Average function enabled
+        ======  =======
+
+        """,
+        validator = strict_discrete_set,
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    vision_balance = Instrument.control(
+        "SOUR:TEL:MOD:VIS:BAL?",
+        "SOUR:TEL:MOD:VIS:BAL %g",
+        """ A float property that controls the balance of the vision modulator
+
+        valid range: -0.5 .. 0.5
+        """,
+        validator = strict_range,
+        values=[-0.5, 0.5],
+        )
+
+    vision_clamping_average = Instrument.control(
+        "SOUR:TEL:MOD:VIS:CLAM:AVER?",
+        "SOUR:TEL:MOD:VIS:CLAM:AVER %g",
+        """ A float property that controls the operation point of the vision modulator
+
+        valid range: -0.5 .. 0.5
+        """,
+        validator = strict_range,
+        values=[-0.5, 0.5],
+        )
+
+    vision_clamping_enabled = Instrument.control(
+        "SOUR:TEL:MOD:VIS:CLAM:STAT?",
+        "SOUR:TEL:MOD:VIS:CLAM:STAT %s",
+        """ A bool property that controls the clamping behavior of the vision modulator,
+
+        ======  =======
+        Value   Meaning
+        ======  =======
+        False   Clamping disabled
+        True    Clamping enabled
+        ======  =======
+
+        """,
+        validator = strict_discrete_set,
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    vision_clamping_mode = Instrument.control(
+        "SOUR:TEL:MOD:VIS:CLAM:TYPE?",
+        "SOUR:TEL:MOD:VIS:CLAM:TYPE %s",
+        """ A string property that controls the clamping mode of the vision modulator
+
+        Possible selections are HARD or SOFT
+
+        """,
+        validator = strict_discrete_set,
+        values=["HARD","SOFT"],
+        )
+
+
+    vision_precorrection_enabled = Instrument.control(
+        "SOUR:TEL:MOD:VIS:PREC?",
+        "SOUR:TEL:MOD:VIS:PREC %s",
+        """ A bool property that controls the precorrection behavior of the vision modulator
+
+        ======  =======
+        Value   Meaning
+        ======  =======
+        False   Precorrection disabled
+        True    Precorrection enabled
+        ======  =======
+
+        """,
+        validator = strict_discrete_set,
+        values={False:0, True:1},
+        map_values = True,
+        )
+
+    vision_residual_carrier_level = Instrument.control(
+        "SOUR:TEL:MOD:VIS:RES?",
+        "SOUR:TEL:MOD:VIS:RES %g",
+        """ A float property that controls the value of the residual carrier
+
+        valid range: 0 .. 0.3 (30%)
+        """,
+        validator = strict_range,
+        values=[-0, 0.3],
```

Comment:
is the "-0" intentional?

Reply:
No, that was propbly a leftover from copy&paste...
Fixed